### PR TITLE
XPANSE Marketplace

### PR DESCRIPTION
**Pull request by daryakoval**
Adding Xpanse marketplace.
Jira ticket: https://jira-hq.paloaltonetworks.local/browse/CIAC-3587
Waiting for demisto-sdk: https://github.com/demisto/demisto-sdk/pull/2460

- [x] Added "xpanse" tag to the marketplace in pack_metadata for packs that we need.
- [x] Added "prepare-testing-bucket-xpanse" step to the build and to the upload-flow
- [x] Changed demisto-sdk version in content repo

Upload:
- [x] Added support for the configuration variables of the XPANSE marketplase buckets in upload flow.
- [x] Added the "upload-packs-to-xpanse-marketplace" step to the upload-flow.
- [x] Added the XPANSE result to the slack message.

### CODE REVIEW

**Review by dantavori — APPROVED** [review]
collect tests files need review

File: Tests/scripts/collect_tests/collect_tests.py [hunk 1]
```diff
@@ -272,8 +273,7 @@ def sanity_tests(self) -> Optional[CollectionResult]:
 
     @property
     def _always_installed_packs(self) -> Optional[CollectionResult]:
-        always_installed_packs_list = ALWAYS_INSTALLED_PACKS_MARKETPLACE_V2 if \
-            self.marketplace == MarketplaceVersions.MarketplaceV2 else ALWAYS_INSTALLED_PACKS_XSOAR
+        always_installed_packs_list = ALWAYS_INSTALLED_PACKS_MAPPING.get(self.marketplace, ALWAYS_INSTALLED_PACKS_XSOAR)
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:276]: in this case, better not use a safe getter

```suggestion
        always_installed_packs_list = ALWAYS_INSTALLED_PACKS_MAPPING[self.marketplace]
```

File: Tests/scripts/collect_tests/collect_tests.py [hunk 2]
```diff
@@ -287,6 +287,8 @@ def _sanity_test_names(self) -> tuple[str, ...]:
                 return tuple(self.conf['test_marketplacev2'])
             case MarketplaceVersions.XSOAR:
                 return XSOAR_SANITY_TEST_NAMES
+            case MarketplaceVersions.XPANSE:
+                return tuple()
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:291]:
```suggestion
                return ()  # none at the moment
```

File: Tests/scripts/collect_tests/collect_tests.py [hunk 3]
```diff
@@ -507,14 +509,14 @@ def __validate_marketplace_compatibility(self,
 
         match self.marketplace:
             case MarketplaceVersions.MarketplaceV2:
-                if content_item_marketplaces != (self.marketplace,):
-                    # marketplacev2 must be the only value in order to be collected
+                # For XSIAM machines we collect tests that have not xsoar marketplace.
+                # Tests for the packs that has only mpv2, or mpv2 and xpanse marketplaces,
+                # will run on xsiam machines only.
+                if MarketplaceVersions.XSOAR in content_item_marketplaces:
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:515]: If content has `xpanse` only, the existing code would consider it as XSIAM compatible. Is it so?
If not, please accept the following suggestion 

```suggestion
                if (MarketplaceVersions.MarketplaceV2 not in content_item_marketplaces) or (MarketplaceVersions.XSOAR in content_item_marketplaces):
```

File: Tests/scripts/collect_tests/collect_tests.py [hunk 4]
```diff
@@ -1029,6 +1031,14 @@ def output(result: Optional[CollectionResult]):
     PATHS.output_machines_file.write_text(json.dumps({str(machine): (machine in machines) for machine in Machine}))
 
 
+class XPANSENightlyTestCollector(NightlyTestCollector):
+    def __init__(self, graph: bool = False):
+        super().__init__(MarketplaceVersions.XPANSE, graph=graph)
+
+    def _collect(self) -> Optional[CollectionResult]:
+        return None
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:1039]:
```suggestion
        logger.info('tests are not currently supported for XPANSE, returning nothing.')
        return None
```

**Review by dorschw — COMMENTED** [review]
👍

File: .gitlab/ci/on-push.yml [hunk 5]
```diff
@@ -194,6 +194,66 @@ prepare-testing-bucket-mpv2:
...
+    - cp "./Tests/conf.json" "$ARTIFACTS_FOLDER/conf.json"
+    - section_end "Create Content Artifacts and Update Conf"
+
+    - section_start "Calculate Packs Dependencies" --collapsed
+    - demisto-sdk find-dependencies -idp Tests/id_set.json --all-packs-dependencies -o $ARTIFACTS_FOLDER/packs_dependencies.json
+    - section_end "Calculate Packs Dependencies"
+
+    - section_start "Replace Cortex XSOAR" --collapsed
+    - pushd "$ARTIFACTS_FOLDER"
+    - export PRODUCT_NAME_XPANSE="XPANSE"
```
Comment (daryakoval) [inline .gitlab/ci/on-push.yml:231]:
```suggestion
    - export PRODUCT_NAME_XPANSE="Cortex XPANSE"
```

**Pull request merged**